src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/Utils.java

```diff
@@ -328,6 +328,53 @@ public boolean isAnnotated(TypeMirror e) {
         return !e.getAnnotationMirrors().isEmpty();
     }
 
+    public boolean isRecursivelyAnnotated(TypeMirror e) {
+        return new SimpleTypeVisitor14<Boolean, Void>() {
+            @Override
+            protected Boolean defaultAction(TypeMirror e, Void unused) {
+                return isAnnotated(e);
+            }
+
+            @Override
+            public Boolean visitArray(ArrayType t, Void unused) {
+                // kind 0
```

Comment:
Just wondering: what does "kind n" refer to?

Reply:
It is the "type path kind" as defined in https://docs.oracle.com/javase/specs/jvms/se15/html/jvms-4.html#jvms-4.7.20.2-210